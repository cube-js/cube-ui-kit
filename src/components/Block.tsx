import { forwardRef } from 'react';
import {
  AllBaseProps,
  CONTAINER_STYLES,
  ContainerStyleProps,
  extractStyles,
  filterBaseProps,
  tasty,
} from '../tasty';

const BlockElement = tasty({
  styled: {
    display: 'block',
  },
});

export interface CubeBlockProps
  extends Omit<AllBaseProps, keyof ContainerStyleProps | 'as'>,
    ContainerStyleProps {}

export const Block = forwardRef((props: CubeBlockProps, ref) => {
  const styles = extractStyles(props, CONTAINER_STYLES);

  return (
    <BlockElement
      {...filterBaseProps(props, { eventProps: true })}
      styles={styles}
      ref={ref}
    />
  );
});
