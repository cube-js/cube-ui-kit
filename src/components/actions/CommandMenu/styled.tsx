import { tasty } from '../../../tasty';

export const StyledCommandMenu = tasty({
  qa: 'CommandMenu',
  styles: {
    display: 'grid',
    flow: 'row',
    gridColumns: 'minmax(0, 1fr)',
    gridRows: {
      '': 'max-content minmax(0, 1fr)',
      header: 'max-content max-content minmax(0, 1fr)',
      footer: 'max-content minmax(0, 1fr) max-content max-content',
      'header & footer':
        'max-content max-content minmax(0, 1fr) max-content max-content',
    },
    placeContent: 'stretch',
    placeItems: 'stretch',
    fill: '#white',
    border: '#border',
    radius: '(1cr + 1bw)',
    shadow: {
      '': false,
      'popover | tray': '0px 5px 15px #dark.05',
    },
    overflow: 'hidden',
    height: {
      '': 'initial',
      popover: 'initial max-content (50vh - $size-md)',
      tray: 'initial max-content (100vh - $size-md)',
    },
  },
});

export const StyledSearchInput = tasty({
  qa: 'SearchInput',
  as: 'input',
  styles: {
    display: 'grid',
    width: '100%',
    color: '#dark',
    fill: '#clear',
    border: '#border bottom',
    outline: 'none',
    transition: 'theme',
    radius: 0,
    textAlign: 'left',
    reset: 'input',
    preset: 't3',
    margin: 0,
    boxSizing: 'border-box',
    userSelect: 'auto',
    height: '($size + 1x)',
    padding: {
      '': '.5x $inline-padding',
      prefix: '0 $inline-padding 0 .5x',
    },

    $size: {
      '': '$size-md',
      '[data-size="small"]': '$size-sm',
      '[data-size="medium"]': '$size-md',
      '[data-size="large"]': '$size-lg',
    },
    '$inline-padding':
      'max($min-inline-padding, (($size - 1lh) / 2 + $inline-compensation - 1bw))',
    '$inline-compensation': '1x',
    '$min-inline-padding': '1x',
  },
});

export const StyledLoadingWrapper = tasty({
  qa: 'LoadingWrapper',
  styles: {
    display: 'flex',
    padding: '2x',
    placeContent: 'center',
    placeItems: 'center',
    color: '#dark-03',
  },
});

export const StyledEmptyState = tasty({
  qa: 'EmptyState',
  styles: {
    display: 'flex',
    padding: '2x',
    placeContent: 'center',
    placeItems: 'center',
    color: '#dark-03',
    preset: 't3',
  },
});

export const StyledMenuWrapper = tasty({
  qa: 'MenuWrapper',
  styles: {
    display: 'grid',
    flow: 'row',
    gridColumns: 'minmax(0, 1fr)',
    placeContent: 'stretch',
    placeItems: 'stretch',
    width: '100%',
    overflow: 'auto',
    scrollbar: 'styled',
  },
});
