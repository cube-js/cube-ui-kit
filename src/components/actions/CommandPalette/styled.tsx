import { tasty } from '../../../tasty';

export const StyledCommandPalette = tasty({
  qa: 'CommandPalette',
  styles: {
    display: 'grid',
    flow: 'row',
    gridColumns: '1fr',
    gridRows: 'auto minmax(0, 1fr)',
    fill: '#white',
    border: '#border',
    radius: '(1cr + 1bw)',
    boxShadow: '0px 5px 15px #dark.05',
    overflow: 'hidden',
    minWidth: '20x',
    maxWidth: '50x',
    maxHeight: '40x',
  },
});

export const StyledSearchWrapper = tasty({
  qa: 'SearchWrapper',
  styles: {
    display: 'flex',
    flow: 'row',
    align: 'center',
    padding: '1x',
    border: '#border bottom',
    fill: '#white',
    gap: '.75x',
  },
});

export const StyledSearchInput = tasty({
  qa: 'SearchInput',
  as: 'input',
  styles: {
    display: 'flex',
    flex: 1,
    border: 'none',
    outline: 'none',
    fill: 'transparent',
    color: '#dark',
    preset: 't3',
    padding: '0',

    '&::placeholder': {
      color: '#dark-03',
    },
  },
});

export const StyledLoadingWrapper = tasty({
  qa: 'LoadingWrapper',
  styles: {
    display: 'flex',
    padding: '2x',
    placeContent: 'center',
    placeItems: 'center',
    color: '#dark-03',
  },
});

export const StyledEmptyState = tasty({
  qa: 'EmptyState',
  styles: {
    display: 'flex',
    padding: '2x',
    placeContent: 'center',
    placeItems: 'center',
    color: '#dark-03',
    preset: 't3',
  },
});

export const StyledMenuWrapper = tasty({
  qa: 'MenuWrapper',
  styles: {
    display: 'grid',
    flow: 'row',
    gridColumns: 'minmax(0, 1fr)',
    placeContent: 'stretch',
    placeItems: 'stretch',
    width: '100%',
    overflow: 'auto',
    scrollbar: 'styled',
  },
});
