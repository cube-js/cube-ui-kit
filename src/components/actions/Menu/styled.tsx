import { DEFAULT_BUTTON_STYLES, DEFAULT_NEUTRAL_STYLES } from '..';
import { tasty } from '../../../tasty';
import { Space } from '../../layout/Space';

export const StyledMenuWrapper = tasty({
  qa: 'Menu',
  styles: {
    display: 'grid',
    gridRows: 'max-content 1sf max-content',
    fill: '#white',
    margin: 0,
    padding: 0,
    border: true,
    radius: '(1cr + 1bw)',
    shadow: {
      '': '',
      'popover | tray': '0px 5px 15px #dark.05',
    },
    height: {
      '': 'initial',
      popover: 'initial max-content (50vh - 4x)',
      tray: 'initial max-content (100vh - 4x)',
    },
    boxSizing: 'border-box',
  },
});

export const StyledMenu = tasty({
  as: 'ul',
  qa: 'MenuList',
  styles: {
    display: 'flex',
    flow: 'column',
    gap: {
      '': '1bw',
      sections: false,
    },
    boxSizing: 'border-box',
    margin: 0,
    padding: {
      '': '0.5x',
      section: 0, // section menu
    },
    overflow: {
      '': 'auto',
      section: '',
    },
    scrollbar: 'styled',
  },
});

export const StyledDivider = tasty({
  qa: 'Divider',
  as: 'li',
  styles: {
    display: 'flex',
    flow: 'column',
    margin: '.5x',
    listStyle: 'none',
    fill: '#border',
    height: '1bw',
    flexShrink: 0,
  },
});

export const StyledHeader = tasty(Space, {
  qa: 'Header',
  as: 'div',
  role: 'heading',
  'aria-level': 3,
  styles: {
    color: '#dark-02',
    preset: 't3',
    placeContent: 'space-between',
    placeItems: 'center',
    whiteSpace: 'nowrap',
    padding: '.5x 1.5x',
    height: {
      '': 'min 4x',
      '[data-size="medium"]': 'min 5x',
    },
    boxSizing: 'border-box',
    border: 'bottom',
  },
});

export const StyledFooter = tasty(Space, {
  qa: 'Footer',
  as: 'div',
  role: 'footer',
  styles: {
    color: '#dark-02',
    preset: 't3',
    placeContent: 'space-between',
    placeItems: 'center',
    whiteSpace: 'nowrap',
    padding: '.5x 1.5x',
    height: {
      '': 'min 4x',
      '[data-size="medium"]': 'min 5x',
    },
    boxSizing: 'border-box',
    border: 'top',
  },
});

export const StyledSection = tasty({
  qa: 'Section',
  as: 'li',
  styles: {
    display: 'flex',
    flow: 'column',
    margin: '0',
    padding: '0',
    listStyle: 'none',
    fill: '#white',
  },
});

export const StyledItem = tasty({
  qa: 'Item',
  as: 'li',
  styles: {
    // Base button-like styles merged from Action/Button defaults
    ...DEFAULT_BUTTON_STYLES,
    ...DEFAULT_NEUTRAL_STYLES,

    // Override specifics for menu context
    display: 'flex',
    flow: 'row',
    justifyContent: 'stretch',
    listStyle: 'none',
    flexShrink: 0,
    height: {
      '': 'min 4x',
      '[data-size="medium"]': 'min 5x',
    },
    border: false,
    boxSizing: 'border-box',
    fill: {
      '': '#clear',
      focused: '#dark.03',
      selected: '#dark.06',
      'selected & focused': '#dark.09',
      pressed: '#dark.06',
      disabled: '#clear',
    },
    color: {
      '': '#dark-02',
      'selected | pressed': '#dark',
      disabled: '#dark-04',
    },
    cursor: {
      '': 'default',
      disabled: 'not-allowed',
    },
    shadow: '#clear',
    padding: {
      '': '.5x 1x',
      'selectionIcon & selectable & !selected': '.5x 1x .5x (1x + 3x)',
    },
    gap: '.75x',
    outline: false,

    // Sub-elements reused from MenuButton styles
    ButtonIcon: {
      display: 'grid',
      fontSize: '@icon-size',
      width: '@icon-size',
      height: '@icon-size',
      placeSelf: 'center',
      placeItems: 'center',
    },

    Postfix: {
      color: {
        '': '#dark-03',
        pressed: '#dark-02',
        disabled: '#dark-04',
      },
    },

    Description: {
      preset: 't4',
      color: '#dark-03',
    },
  },
});

export const StyledSectionHeading = tasty(Space, {
  qa: 'SectionHeading',
  as: 'div',
  styles: {
    color: '#dark-04',
    preset: 'c2',
    padding: '.5x 1x',
    height: '3x',
    placeContent: 'center space-between',
    align: 'start',
  },
});
