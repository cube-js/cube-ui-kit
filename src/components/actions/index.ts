import { Button as __Button } from './Button';
import { ButtonGroup } from './ButtonGroup/ButtonGroup';

const Button = Object.assign(
  __Button as typeof __Button & {
    Group: typeof ButtonGroup;
  },
  { Group: ButtonGroup },
);

export * from './Button';
export * from './Action';
export * from './use-action';
export { Button, ButtonGroup };
