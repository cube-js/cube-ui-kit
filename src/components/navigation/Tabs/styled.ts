import { Styles, tasty } from '../../../tasty';

export const StyledTabsContainer = tasty({});

export const StyledTabPanes = tasty({
  styles: {
    display: 'flex',
    flow: 'row',
    gap: '3x',
  },
});

export const StyledTabItem = tasty({
  styles: {
    preset: 'h5s',
    display: 'flex',
    placeItems: 'center stretch',
    placeContent: 'center',
    flow: 'row',
    gap: '1x',
    fill: '#clear',
    color: {
      '': '#dark',
      selected: '#purple-text',
      hovered: '#purple-text',
    },
    outline: {
      '': '#purple-03.0',
      focused: '#purple-03',
    },
    border: {
      '': 0,
      selected: 'bottom 3bw #purple-text',
    },
    cursor: {
      '': 'pointer',
      disabled: 'default',
    },
    padding: {
      '': '1.5x 0',
    },
    radius: {
      '': '1r',
      selected: '1r 1r 0 0',
    },
  },
});

export const StyledTabBody = tasty({});

export const ACTION_BUTTON: Styles = {
  border: {
    '': '#clear',
    pressed: '#clear',
  },
  fill: {
    '': '#clear',
    hovered: '#clear',
    'pressed | selected': '#clear',
    disabled: '#clear',
  },
  color: {
    '': '#dark-02',
    hovered: '#dark-02',
    'pressed | selected': '#purple-text',
    disabled: '#dark-04',
  },
  padding: {
    '': '1.5x 0',
  },
  cursor: {
    '': 'pointer',
    disabled: 'default',
  },
  shadow: '#clear',
  display: 'flex',
  flow: 'row',
  justifyContent: 'start',
};
