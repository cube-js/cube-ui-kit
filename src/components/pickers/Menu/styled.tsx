import { tasty } from '../../../tasty';
import { Space } from '../../layout/Space';

export const StyledMenu = tasty({
  as: 'ul',
  qa: 'Menu',
  styles: {
    display: 'flex',
    flow: 'column',
    gap: {
      '': '1bw',
      sections: '',
    },
    fill: '#white',
    margin: '0',
    padding: {
      '': '0.5x',
      section: '0.5x', // section menu
      sections: '0', // has sections inside
    },
    overflow: {
      '': 'hidden',
      section: '',
    },
    border: {
      '': '#dark-05',
      section: '',
    },
    radius: {
      '': '1r',
    },
    boxShadow: {
      '': '',
      popover: '0px 5px 15px #dark.05',
    },
  },
});

export const StyledDivider = tasty({
  as: 'li',
  styles: {
    display: 'flex',
    flow: 'column',
    margin: '0 -0.5x',
    listStyle: 'none',
    fill: '#dark.05',
    height: '1bw',
  },
});

export const StyledMenuHeader = tasty(Space, {
  as: 'li',
  styles: {
    fill: '#light',
    color: '#dark-02',
    preset: 't2m',
    padding: '0.75x 2x',
    margin: '-0.5x -0.5x (0.5x - 1bw) -0.5x',
    border: '#dark-05 bottom',
    placeContent: 'space-between',
    align: 'start',
    radius: '1r 1r 0 0',
  },
});

export const StyledMenuSection = tasty({
  as: 'li',
  styles: {
    display: 'flex',
    flow: 'column',
    margin: '0',
    padding: '0',
    listStyle: 'none',
    fill: '#white',
  },
});

export const StyledMenuItem = tasty({
  as: 'li',
  styles: {
    display: 'flex',
    flow: 'column',
    gap: '1bw',
    margin: '0',
    padding: '0',
    listStyle: 'none',
    fill: '#white',
    preset: 't3m',
    color: {
      '': 'inherit',
      disabled: '#dark-04',
    },
  },
});

export const StyledMenuSectionHeading = tasty(Space, {
  as: 'header',
  styles: {
    color: '#dark-03',
    fill: '#light',
    preset: 'c2',
    padding: '(1x - 1bw) 2x',
    placeContent: 'space-between',
    align: 'start',
  },
});
