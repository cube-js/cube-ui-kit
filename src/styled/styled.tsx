import styledComponents, { createGlobalStyle } from 'styled-components';
import { ComponentType, FC, forwardRef, useContext, useMemo } from 'react';
import { isValidElementType } from 'react-is';
import { BreakpointsContext } from '../providers/BreakpointsProvider';
import { modAttrs } from '../utils/react';
import { useContextStyles } from '../providers/StylesProvider';
import {
  AllBaseProps,
  BaseStyleProps,
  StyledProps,
  GlobalStyledProps,
} from '../components/types';
import { renderStyles } from '../utils/renderStyles';
import { pointsToZones } from '../utils/responsive';
import { Styles, StylesInterface } from '../styles/types';
import { BASE_STYLES } from '../styles/list';
import { ResponsiveStyleValue } from '../utils/styles';
import { mergeStyles } from '../utils/mergeStyles';
import { deprecationWarning } from '../utils/warnings';

export type AllBasePropsWithMods<K extends (keyof StylesInterface)[]> =
  AllBaseProps & {
    [key in K[number]]?: ResponsiveStyleValue<StylesInterface[key]>;
  } & BaseStyleProps;

type StyledPropsWithDefaults<
  Props extends { styles?: Styles },
  DefaultProps extends Partial<Props>,
> = keyof DefaultProps extends never
  ? Props
  : {
      [key in Extract<keyof Props, keyof DefaultProps>]?: Props[key];
    } & {
      [key in keyof Omit<Props, keyof DefaultProps>]: Props[key];
    };

type EitherLegacyPropsOrInlined<
  K extends (keyof StylesInterface)[],
  DefaultProps,
> =
  | ((Pick<StyledProps<K, DefaultProps>, 'name' | 'tag' | 'styles'> & {
      /**
       * @deprecated
       */
      props?: DefaultProps;
    }) & {
      [key in keyof DefaultProps]?: 'You Should use either legacy props field or inline props';
    })
  | ({
      props?: 'You Should use either legacy props field or inline props';
    } & DefaultProps);

function styled<
  K extends (keyof StylesInterface)[],
  Props,
  DefaultProps extends Partial<Props> = Partial<Props>,
>(options: StyledProps<K, DefaultProps>, secondArg?: never);
function styled(selector: string, styles?: Styles);
function styled<
  K extends (keyof StylesInterface)[],
  Props extends { styles?: Styles },
  DefaultProps extends Partial<Props> = Partial<Props>,
>(
  Component: ComponentType<Props>,
  options?: EitherLegacyPropsOrInlined<K, DefaultProps>,
): ComponentType<StyledPropsWithDefaults<Props, DefaultProps>>;

// Implementation
function styled<
  K extends (keyof StylesInterface)[],
  C = Record<string, unknown>,
>(Component, options) {
  deprecationWarning(options?.props == null, {
    property: 'props',
    name: 'styled api',
    betterAlternative:
      "inline props directly in styled(), eg: styled({ type: 'confirm' })",
  });

  if (typeof Component === 'string') {
    let selector = Component;
    let styles = options;
    let Element = createGlobalStyle`${(css) => css}`;

    const _Component: FC<GlobalStyledProps> = ({ breakpoints }) => {
      let contextBreakpoints = useContext(BreakpointsContext);
      let zones = pointsToZones(breakpoints || contextBreakpoints);

      let css = `${
        styles ? `\n{}${selector}{${renderStyles(styles, zones)}}` : ''
      }`;

      return <Element css={css} />;
    };

    _Component.displayName = `CubeStyled(${
      Element.displayName ?? Element.name ?? Component
    })`;

    return _Component;
  }

  if (isValidElementType(Component)) {
    let {
      styles: extendStyles,
      props: legacyDefaultProps,
      name,
      tag: extendTag,
      ...defaultProps
    } = options ?? {};

    const _StyledComponent = forwardRef((props: C, ref) => {
      const { styles, styleName, as, ...restProps } =
        props as AllBasePropsWithMods<K>;

      const mergedStyles: Styles | undefined = useMemo(() => {
        if (extendStyles != null && styles != null) {
          return mergeStyles(extendStyles, styles);
        }

        return extendStyles;
      }, [styles]);

      return (
        <Component
          ref={ref}
          {...legacyDefaultProps}
          {...defaultProps}
          {...restProps}
          styles={mergedStyles}
          styleName={styleName ?? name}
          as={as ?? extendTag}
        />
      );
    });

    _StyledComponent.displayName = `CubeStyled(${
      Component.displayName ?? Component.name ?? 'Anonymous'
    })`;

    return _StyledComponent;
  }

  options = Component;

  let {
    name,
    tag = 'div',
    css: defaultCSS,
    styles: defaultStyles,
    props: legacyDefaultProps,
    styleProps,
    ...defaultProps
  } = options;

  let Element = styledComponents[tag](({ css }) => css);

  let _StyledComponent = forwardRef(
    (allProps: AllBasePropsWithMods<K>, ref) => {
      let {
        as,
        styles,
        styleName,
        breakpoints,
        mods,
        element,
        qa,
        qaVal,
        css,
        ...props
      } = allProps;

      let propStyles: Styles = useMemo(
        () =>
          (
            (styleProps
              ? (styleProps as (keyof StylesInterface)[]).concat(BASE_STYLES)
              : BASE_STYLES) as (keyof StylesInterface)[]
          ).reduce((map, prop) => {
            if (prop in props) {
              map[prop] = props[prop];

              delete props[prop];
            }

            return map;
          }, {}),
        [props],
      );

      let contextStyles = useContextStyles(styleName ?? name, props);
      let allStyles: Styles = useMemo(
        () => mergeStyles(defaultStyles, contextStyles, styles, propStyles),
        [contextStyles, styles, propStyles],
      );

      let contextBreakpoints = useContext(BreakpointsContext);
      let zones = pointsToZones(breakpoints ?? contextBreakpoints);

      css = `${
        typeof defaultCSS === 'function' ? defaultCSS(props) : defaultCSS ?? ''
      }${typeof css === 'function' ? css(props) : css ?? ''}${
        allStyles ? renderStyles(allStyles, zones) : ''
      }`;

      if (mods) {
        Object.assign(props, modAttrs(mods));
      }

      return (
        <Element
          as={as ?? tag}
          data-element={element}
          data-qa={qa}
          data-qaval={qaVal}
          {...legacyDefaultProps}
          {...defaultProps}
          {...props}
          ref={ref}
          css={css}
        />
      );
    },
  );

  _StyledComponent.displayName = `CubeStyled(${
    Element.displayName ?? Element.name ?? tag
  })`;

  return _StyledComponent;
}

export { styled };
