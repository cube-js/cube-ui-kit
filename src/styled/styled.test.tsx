import { getByTestId, render } from '@testing-library/react';

import { Button } from '../components/actions';
import { Block } from '../components/Block';

import { styled } from './styled';

import 'jest-styled-components';
jest.unmock('styled-components');

describe('styled() API', () => {
  beforeAll(() => {
    jest.spyOn(console, 'group').mockImplementation(() => {});
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should warn about deprecated API', () => {
    const spiedConsole = jest
      .spyOn(console, 'warn')
      .mockImplementation(() => {});

    styled(Button, { props: { type: 'primary' } });

    expect(spiedConsole).toBeCalledTimes(1);
    expect(spiedConsole.mock.calls).toMatchSnapshot();
  });

  it('should provide defaults and give ability to override', () => {
    const SButton = styled(Button, { type: 'primary' });

    const { getByTestId, rerender } = render(<SButton qa="button" />);
    expect(getByTestId('button').dataset.type).toBe('primary');

    rerender(<SButton type="secondary" qa="button" />);
    expect(getByTestId('button').dataset.type).toBe('secondary');
  });

  it('should pass styles from styled', () => {
    const StyledBlock = styled(Block, { styles: { color: '#clear.1' } });
    const { container } = render(<StyledBlock />);

    expect(container).toMatchSnapshot();
  });

  it('should be able to override styles', () => {
    const StyledBlock = styled(Block, { styles: { color: '#clear.1' } });
    const { container } = render(
      <StyledBlock styles={{ color: '#black.1' }} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should pass qa prop', () => {
    const StyledBlock = styled({ props: { qa: 'Field' } });
    const { container } = render(<StyledBlock />);

    expect(getByTestId(container, 'Field', {})).toBeDefined();
  });
});
