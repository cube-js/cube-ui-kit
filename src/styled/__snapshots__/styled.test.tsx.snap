// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled() API should be able to override styles 1`] = `
.c0 {
  outline: none;
}

.c0[hidden] {
  display: none !important;
}

.c0 {
  display: block;
}

.c0 {
  color: var(--black-color);
}

@media (min-width:980px) {

}

@media (max-width:979px) {

}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`styled() API should pass styles from styled 1`] = `
.c0 {
  outline: none;
}

.c0[hidden] {
  display: none !important;
}

.c0 {
  display: block;
}

.c0 {
  color: rgba(var(--clear-color-rgb),0.1);
}

@media (min-width:980px) {

}

@media (max-width:979px) {

}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`styled() API should warn about deprecated API 1`] = `
Array [
  Array [
    "CubeUIKit:",
    "\\"props\\" is deprecated, consider better alternative: inline props directly in styled(), eg: styled({ type: 'confirm' })",
  ],
]
`;
